env/iasConfig.go

```diff
@@ -46,17 +52,103 @@ func GetIASConfig() (*Identity, error) {
 			return nil, fmt.Errorf("cannot parse vcap services: %w", err)
 		}
 		if len(vcapServices.Identity) == 0 {
-			return nil, fmt.Errorf("no '" + iasServiceName + "' service instance bound to the application")
+			return nil, fmt.Errorf("no '%s' service instance bound to the application", iasServiceName)
 		}
 		if len(vcapServices.Identity) > 1 {
-			return nil, fmt.Errorf("more than one '" + iasServiceName + "' service instance bound to the application. This is currently not supported")
+			return nil, fmt.Errorf("more than one '%s' service instance bound to the application. This is currently not supported", iasServiceName)
 		}
 		return &vcapServices.Identity[0].Credentials, nil
 	case kubernetes:
-		return nil, fmt.Errorf("unable to parse ias config: kubernetes env detected but not yet supported")
+		var secretPath = os.Getenv(iasConfigPathKey)
+		if secretPath == "" {
+			secretPath = iasConfigPathDefault
+		}
+		identities, err := readServiceBindings(secretPath)
+		if err != nil || len(identities) == 0 {
+			return nil, fmt.Errorf("cannot find '%s' service binding on secret path '%s'", iasServiceName, secretPath)
+		} else if len(identities) > 1 {
+			return nil, fmt.Errorf("found more than one '%s' service instance on secret path '%s'. This is currently not supported", iasServiceName, secretPath)
+		}
+		return &identities[0], nil
 	default:
-		return nil, fmt.Errorf("unable to parse ias config: unknown environment detected")
+		return nil, fmt.Errorf("unable to parse '%s' service config: unknown environment detected", iasServiceName)
+	}
+}
+
+func readServiceBindings(secretPath string) ([]Identity, error) {
+	bindingFiles, err := ioutil.ReadDir(secretPath)
```

Comment:
is this not expected to be directory instead of files? it confuses me, I think without files just bindings it would be better.